docs/conandata_yml_format.md

```diff
@@ -0,0 +1,220 @@
+
+#### portability
+
+`patch_type: portability` indicates the patch improves the portability of the library, e.g. adding supports of new architectures (ARM, Sparc, etc.), operation systems (FreeBSD, Android, etc.), compilers (Intel, MinGW, etc.), and other types of configurations which are not originally supported by the project.
+in such cases, the patch could be adopted from another package repository (e.g. MSYS packages, Debian packages, Homebrew, FreeBSD ports, etc.).
+patches of this kind are preferred to be submitted upstream first (in contrast to patch_type: conan), but it's not always possible.
+some projects simply do not accept patches for platforms they don't have a build/test infrastracture. some projects are just either abandoned or inactive.
+
+#### conan
+
+`patch_type: conan` indicates the patch that is conan-specific, patches of such kind are usually not welcomed upstream at all, because they provide zero value outside of conan.
+examples of such a patches may include modifications of build system files to allow dependencies provided by conan instead of dependencies provided by projects themselves (e.g. as submodule or just 3rd-party sub-directory) or by the system package manager (rpm/deb).
+such patches, for instance, may contain variables are targets generated only by conan, but not generated normally by the build system (e.g. `CONAN_INCLUDE_DIRS`).
+
+### patch_source
+
+`patch_source` is an URL from where patch was taken from, https scheme is preferred, but other URLs (e.g. git/svn/hg) are also accepted if there is no alternative. what could be a patch source?
+
+- link to the public commit in project hosting like GitHub/GitLab/BitBucket/Savanha/SourceForge/etc
+- link to the Pull Request or equivalent (e.g. gerrit review)
+- link to the bug tracker (such as JIRA, BugZilla, etc.)
+- link to the mail list discussion
+- link to the patch itself in another repository (e.g. MSYS, Debian, etc.)
+
+for the `patch_type: portability` there might be no patch source matching the definition above. although we encourage contributors to submit all such portability fixes upstream first, it's not always possible (e.g. for projects no longer maintained), in such case link to the conan issue is a valid patch source (if there is no issue, you may create one).
+for the `patch_type: conan`, it doesn't make sense to submit patch upstream, so there will be no patch source.
+
+### patch_description
+
+`patch_description` is an arbitrary text describing the following aspects of the patch:
+
+- what does patch do (example - `add missing unistd.h header`)
+- why is it necessary (example - `port to Android`)
+- how exactly does patch achieve that (example - `update configure.ac`)
+
+therefore, the full description may look like: `port to Android: update configure.ac adding missing unistd.h header`
+
+## reduce
```

Comment:
Probably we can move this to the FAQ (and link from here). It is something anyone can we interested in: _"How can I get the same recipe-revision when building locally?"_

Reply:
moved section to FAQ